src/readonly-table/index.ts

```diff
@@ -1,220 +1,213 @@
-/* eslint-disable kuc-v1/validator-in-should-update */
 import { html, PropertyValues } from "lit";
-import { property } from "lit/decorators.js";
-import { KucBase } from "../base/kuc-base";
+import { property, state } from "lit/decorators.js";
 import { visiblePropConverter } from "../base/converter";
-import { validateProps } from "../base/validator";
-
-type Column = { header?: { text?: string }; visible?: boolean };
-type ReadOnlyTableProps = {
-  className?: string;
-  id?: string;
-  label?: string;
-  visible?: boolean;
-  columns?: Column[];
-  data?: string[][];
-};
-
-export class ReadOnlyTable extends KucBase {
-  @property({ type: String, reflect: true, attribute: "class" }) className = "";
-  @property({ type: String, reflect: true, attribute: "id" }) id = "";
-  @property({ type: String }) label = "";
-  @property({
-    type: Boolean,
-    attribute: "hidden",
-    reflect: true,
-    converter: visiblePropConverter,
-  })
-  visible = true;
-  @property({ type: Array }) columns: Column[] = [];
-  @property({ type: Array }) data: string[][] = [];
-
-  constructor(props?: ReadOnlyTableProps) {
-    super();
-    if (!props) {
-      return;
+import { ERROR_MESSAGE } from "../base/constant";
+import { createStyleOnHeader, KucBase } from "../base/kuc-base";
+import {
+  throwErrorAfterUpdateComplete,
+  validateColumns,
+  validateData,
+  validateProps,
+  validateRowsPerPage,
+} from "../base/validator";
+import "../base/pagination";
+import { Column, ReadOnlyTableProps } from "./type";
+import { READ_ONLY_TABLE_CSS } from "./style";
+
+let exportReadOnlyTable;
+(() => {
+  exportReadOnlyTable = window.customElements.get("kuc-readonly-table");
+  if (exportReadOnlyTable) {
+    return;
+  }
+  class KucReadOnlyTable extends KucBase {
+    @property({ type: String, reflect: true, attribute: "class" }) className =
+      "";
+    @property({ type: String, reflect: true, attribute: "id" }) id = "";
+    @property({ type: String }) label = "";
+    @property({ type: Array }) columns: Column[] = [];
+    @property({ type: Array }) data: object[] = [];
+    @property({ type: Boolean }) pagination = true;
+    @property({ type: Number }) rowsPerPage = 5;
+    @property({
+      type: Boolean,
+      attribute: "hidden",
+      reflect: true,
+      converter: visiblePropConverter,
+    })
+    visible = true;
+    @state()
+    private _pagePosition: number = 1;
+    @state()
+    private _columnOrder: string[] = [];
+
+    constructor(props?: ReadOnlyTableProps) {
+      super();
+      if (!props) return;
+      const validProps = validateProps(props);
+      Object.assign(this, validProps);
     }
-    if (!Array.isArray(props.columns) && props.columns !== undefined) {
-      throw new Error("'columns' property is invalid");
+
+    shouldUpdate(changedProperties: PropertyValues): boolean {
+      if (changedProperties.has("columns")) {
+        if (!validateColumns(this.columns)) {
```

Comment:
Grouping the condition:
if (changedProperties.has("columns") &&  !validateColumns(this.columns)) {
...